internal/openssh/lib.go

```diff
@@ -0,0 +1,214 @@
+package openssh
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/binary"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"golang.org/x/crypto/ssh"
+)
+
+// This module cherry-picks from https://go-review.googlesource.com/c/crypto/+/218620,
+// that provides a solution for https://github.com/golang/go/issues/37132, that highlights
+// the need for methods to marshal private keys to the OpenSSH format in `x/crypto/ssh`.
+//
+// This provides utilities to serialize a Private Keys in OpenSSH format: once
+// this goes mainstream, we can remove this module.
+
+// Constants represents the algorithm names for key types supported by this package.
+const (
+	KeyAlgoRSA = "ssh-rsa"
```

Comment:
I guess it looks like you're only using them below in the imported code, so I'd probably suggest changing the references to `ssh.KeyAlgoRSA` &c below rather than re-defining them here.